lib/cell.rb

```diff
@@ -0,0 +1,46 @@
+class Cell
+  attr_reader :coordinate,
+              :ship,
+              :render
+
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @ship = nil
+    @fired_upon = false
+    # @render = "."
+  end
+
+  def empty?
+    @ship == nil
+  end
+
+  def place_ship(new_ship)
+    @ship = new_ship
+  end
+
+  def fired_upon?
+    @fired_upon
+  end
+
+  def fire_upon
+    if @ship
+      @ship.hit
+    end
+    @fired_upon = true
+  end
```

Comment:
@sarahgarlock oooh nice job catching an edge case here -- if ship is nil then it shouldn't be fired upon or lose health. love it!

Reply:
@sarahgarlock  I was running this in pry, and logic is solid here for edge case but because we have the @fire_upon in ln 29, it's always going to output true, even if there isn't a ship in there. so we'll want to move ln 29 `@fired_upon = true` up to line 27.5

Reply:
Moved `@fired_upon = true` to ln 27.5